### ISSUE TITLE
`gradle init` fails to generate Kotlin application with JUnit Jupiter test framework

### ISSUE THREAD

**Issue by bamboo** [gradle/gradle#29204]
### Current Behavior

Build fails with:

```
* What went wrong:
Execution failed for task ':init'.
> Could not find template 'kotlinapplication/junitjupiter/AppTest.kt.template' in classpath.
```

### Expected Behavior

Build succeeds

### Steps to Reproduce

Run `gradle init` and choose to generate a single module Kotlin application with Kotlin build scripts and the JUnit Jupiter test framework.

### Gradle version

8.8-rc-1

**Comment by ljacomet** [comment]
This was supposed to be fixed with #28489

**Comment by ov7a** [comment]
Reproducer command:
```sh
gradle init --type kotlin-application --test-framework junit-jupiter --dsl kotlin --project-name test --incubating --java-version 17 --no-split-project
```

**Comment by ljacomet** [comment]
Fixed with #29240
